subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java

```diff
@@ -137,8 +139,10 @@ public void failWith(Throwable err) {
     }
 
     public void restart() {
-        removeFromTargetConfigurations();
-        attachToTargetConfigurations();
+        if (from.isSelected()) {
```

Comment:
If I read this PR correctly for production code, this change is the fix. All the rest are style or performance related changes. If that's the case, is it possible to isolate the fix in its own commit?

Reply:
Splitted into 2 commits.

Reply:
Great